STX.REST.RESTFulSense.Clients/Services/Foundations/HttpExchanges/HttpExchangeService.Validations.cs

```diff
@@ -0,0 +1,130 @@
+// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Net;
+using System.Net.Http;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges.Exceptions;
+
+namespace STX.REST.RESTFulSense.Clients.Services.Foundations.HttpExchanges
+{
+    internal partial class HttpExchangeService
+    {
+        private static void ValidateHttpExchange(
+            HttpExchange httpExchange,
+            HttpMethod defaultHttpMethod,
+            Version defaultHttpVersion,
+            HttpVersionPolicy defaultHttpVersionPolicy)
+        {
+            ValidateHttpExchangeNotNull(httpExchange);
+            ValidateHttpExchangeRequestNotNull(httpExchange.Request);
+
+            ValidateHttpExchangeRequestNotInvalid(
+                httpExchange.Request,
+                defaultHttpMethod,
+                defaultHttpVersion);
+
+            ValidateHttpExchangRequestHeaders(httpExchange.Request.Headers);
+        }
```

Comment:
This validation looks a bit messy. I'm expecting this format...  Every object starts with the null check and then the validation rules...

```suggestion
        private static void ValidateOnGet(
            HttpExchange httpExchange,
            HttpMethod httpMethod,
            Version httpVersion,
            HttpVersionPolicy httpVersionPolicy)
        {
            ValidateHttpExchange(httpExchange);
            ValidateHttpMethod(httpMethod);
            ValidateVersion(httpVersion);
            ValidateHttpVersionPolicy(httpVersionPolicy);
        }

        private static void ValidateHttpExchange(HttpExchange httpExchange)
        {
            ValidateHttpExchangeNotNull(httpExchange);
            ValidateHttpExchangeRequest(httpExchangeRequest: httpExchange.Request);
        }

        private static void ValidateHttpExchangeRequest(HttpExchangeRequest httpExchangeRequest)
        {
            ValidateHttpExchangeRequestNotNull(httpExchangeRequest);

            ValidateHttpRequest(
                (Rule: IsInvalid(httpExchangeRequest.BaseAddress),
                    Parameter: nameof(HttpExchangeRequest.BaseAddress)),

                (Rule: IsInvalid(httpExchangeRequest.RelativeUrl),
                    Parameter: nameof(HttpExchangeRequest.RelativeUrl)),

                . . .
                    
                    );

            ValidateHttpExchangeRequestHeaders(headers: httpExchangeRequest.Headers)
            ValidateHttpExchangeRequestContent(content: httpExchangeRequest.Content)
        }


        private static void ValidateHttpExchangeRequestHeaders(HttpExchangeRequestHeaders headers)
        {
            ValidateHttpExchangeRequestHeadersNotNull(headers);

            ValidateHttpExchangeRequestHeaders
                (Rule: IsInvalid(...),
                    Parameter: nameof(...)));
        }

        private static void ValidateHttpExchangeRequestContent(HttpExchangeContent content)
        {
            ValidateHttpExchangeRequestContentNotNull(headers);

            ValidateHttpExchangeRequestContent
                (Rule: IsInvalid(...),
                    Parameter: nameof(...)));
        }

```

Reply:
Why ValidateOnGet instead ValidateHttpExchange?

Using this input parameters
HttpMethod httpMethod,
Version httpVersion,
HttpVersionPolicy httpVersionPolicy

seems lost the meaning? I mean, What´s the point if it uses what it´s inside httpExchange? using the prefix "default" helps to know what´s the purpose for each input parameter. If "default" doesn´t like to you, maybe "required", "expected" or another prefix could be used.


Why this is appart from the request validation?


ValidateHttpMethod(httpMethod);
ValidateVersion(httpVersion);
ValidateHttpVersionPolicy(httpVersionPolicy);